src/fetch-or-populate-file-cache.test.ts

```diff
@@ -0,0 +1,261 @@
+import { readJsonFile, writeJsonFile } from '@metamask/utils/node';
+import path from 'path';
+
+import { fetchOrPopulateFileCache } from './fetch-or-populate-file-cache';
+import { withinSandbox, fakeDateOnly } from '../tests/helpers';
+
+describe('fetchOrPopulateFileCache', () => {
+  beforeEach(() => {
+    fakeDateOnly();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  describe('if the given file does not already exist', () => {
+    it('saves the return value of the given function in the file as JSON along with its created time', async () => {
+      jest.setSystemTime(new Date('2023-01-01T00:00:00Z'));
+
+      await withinSandbox(async ({ directoryPath: sandboxDirectoryPath }) => {
+        const filePath = path.join(sandboxDirectoryPath, 'cache');
+        const data = { foo: 'bar' };
+
+        await fetchOrPopulateFileCache({
+          filePath,
+          getDataToCache: () => data,
+        });
+
+        const cache = await readJsonFile(filePath);
+        expect(cache).toStrictEqual({
+          ctime: '2023-01-01T00:00:00.000Z',
+          data,
+        });
+      });
+    });
+
+    it('returns the data that was cached', async () => {
+      await withinSandbox(async ({ directoryPath: sandboxDirectoryPath }) => {
+        const filePath = path.join(sandboxDirectoryPath, 'cache');
+        const dataToCache = { foo: 'bar' };
+
+        const cachedData = await fetchOrPopulateFileCache({
+          filePath,
+          getDataToCache: () => dataToCache,
+        });
+
+        expect(cachedData).toStrictEqual(dataToCache);
+      });
+    });
+  });
+
+  describe('if the given file already exists', () => {
+    describe('and no explicit max age is given', () => {
+      describe('and it was created less than an hour ago', () => {
```

Comment:
Nit: Have you considered parameterize these tests? We might be able to cut down on the repetition that way. These all look like they're testing the same thing with different inputs.